### PR TITLE
fix: Correct null_count in describe()

### LINKED ISSUE #10231: null_count in describe() does not show zero when there is no null value in the array
### Describe the bug

Hello, I'm using the input file `datafusion/core/tests/data/cars.csv` and running the `describe()` function. Below is the output of the `describe()` function: 

```
+------------+-------+-------------------+---------------------+
| describe   | car   | speed             | time                |
+------------+-------+-------------------+---------------------+
| count      | 25    | 25.0              | 25                  |
| null_count | 25    | 25.0              | 25                  |
| mean       | null  | 11.84             | null                |
| std        | null  | 6.456456716600317 | null                |
| min        | green | 0.0               | 1996-04-12T12:05:03 |
| max        | red   | 21.5              | 1996-04-12T12:05:15 |
| median     | null  | 11.0              | null                |
+------------+-------+-------------------+---------------------+
``` 

I'm wondering, shouldn't the `null_count` be `0` if there are no null values in the array? Could this be considered a bug?

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Closes #10231

### CODE REVIEW

File: datafusion/expr/src/expr_fn.rs [hunk 1]
```diff
@@ -215,6 +215,18 @@ pub fn count(expr: Expr) -> Expr {
     ))
 }
 
+/// Create an expression to represent the count_null() aggregate function
```
Comment (alamb) [inline datafusion/expr/src/expr_fn.rs:218]: I think this function is creating a function like `COUNT(IS NULL x)` rather than a `count_null` function

File: datafusion/core/src/dataframe/mod.rs [hunk 2]
```diff
@@ -534,7 +534,7 @@ impl DataFrame {
                 vec![],
                 original_schema_fields
                     .clone()
-                    .map(|f| count(is_null(col(f.name()))).alias(f.name()))
+                    .map(|f| count_null(col(f.name())).alias(f.name()))
```
Comment (alamb) [inline datafusion/core/src/dataframe/mod.rs:537]: Instead of a `count_null()` new function, what do you think about using the existing [`count`](https://docs.rs/datafusion/latest/datafusion/logical_expr/fn.count.html) function?

Something like `count(expr.is_null())`

```suggestion
                    .map(|f| count(col(f.name()).is_null()).alias(f.name())))
```

File: datafusion/core/tests/dataframe/describe.rs [hunk 3]
```diff
@@ -39,7 +39,7 @@ async fn describe() -> Result<()> {
...
         "+------------+-------------------+----------+--------------------+--------------------+--------------------+--------------------+--------------------+--------------------+-----------------+------------+-------------------------+--------------------+-------------------+",
         "| count      | 7300.0            | 7300     | 7300.0             | 7300.0             | 7300.0             | 7300.0             | 7300.0             | 7300.0             | 7300            | 7300       | 7300                    | 7300.0             | 7300.0            |",
-        "| null_count | 7300.0            | 7300     | 7300.0             | 7300.0             | 7300.0             | 7300.0             | 7300.0             | 7300.0             | 7300            | 7300       | 7300                    | 7300.0             | 7300.0            |",
+        "| null_count | 0.0               | 0        | 0.0                | 0.0                | 0.0                | 0.0                | 0.0                | 0.0                | 0               | 0          | 0                       | 0.0                | 0.0               |",
```
Comment (alamb) [inline datafusion/core/tests/dataframe/describe.rs:42]: that certainly looks better

**Review by alamb — COMMENTED** [review]
Thank yoU @Weijun-H  -- this is a nice improvement

Would it be possible to add / update a test that shows an output with null count greater than zero as well?

I also left some comments about potential improvements

File: datafusion/core/src/dataframe/mod.rs (hunk 2 above)
Comment (Weijun-H) [reply datafusion/core/src/dataframe/mod.rs:537]: When using `count`, it sums all `non-null` Expr while `is_null` converts `Expr` to a boolean expression, which is why it doesn't work.

File: datafusion/expr/src/expr_fn.rs [hunk 4]
```diff
@@ -215,6 +215,18 @@ pub fn count(expr: Expr) -> Expr {
     ))
 }
 
+/// Create an expression to represent the COUNT(IS NULL x) aggregate function
```
Comment (alamb) [inline datafusion/expr/src/expr_fn.rs:218]: I found this confusing as first because it isn't `COUNT(x IS NULL)` -- I think a more accurate description is `COUNT(x) FILTER (WHERE x IS NULL)` or something like that 

```suggestion
/// Create an aggregate that returns the number `NULL` values in a column
```

**Review by alamb — COMMENTED** [review]
Thanks @Weijun-H - I think this PR needs a test for data that has actual nulls in it

I checked the file the test is using and I don't think it has any null values in it

```sql
> select * from './parquet-testing/data/alltypes_tiny_pages.parquet';
+-----+----------+-------------+--------------+---------+------------+-----------+--------------------+-----------------+------------+-------------------------+------+-------+
| id  | bool_col | tinyint_col | smallint_col | int_col | bigint_col | float_col | double_col         | date_string_col | string_col | timestamp_col           | year | month |
+-----+----------+-------------+--------------+---------+------------+-----------+--------------------+-----------------+------------+-------------------------+------+-------+
| 122 | true     | 2           | 2            | 2       | 20         | 2.2       | 20.2               | 01/13/09        | 2          | 2009-01-13T01:02:05.410 | 2009 | 1     |
| 123 | false    | 3           | 3            | 3       | 30         | 3.3       | 30.299999999999997 | 01/13/09        | 3          | 2009-01-13T01:03:05.430 | 2009 | 1     |
| 124 | true     | 4           | 4            | 4       | 40         | 4.4       | 40.4               | 01/13/09        | 4          | 2009-01-13T01:04:05.460 | 2009 | 1     |
| 125 | false    | 5           | 5            | 5       | 50         | 5.5       | 50.5               | 01/13/09        | 5          | 2009-01-13T01:05:05.500 | 2009 | 1     |
| 126 | true     | 6           | 6            | 6       | 60         | 6.6       | 60.599999999999994 | 01/13/09        | 6          | 2009-01-13T01:06:05.550 | 2009 | 1     |
| 127 | false    | 7           | 7            | 7       | 70         | 7.7       | 70.7               | 01/13/09        | 7          | 2009-01-13T01:07:05.610 | 2009 | 1     |
| 128 | true     | 8           | 8            | 8       | 80         | 8.8       | 80.8               | 01/13/09        | 8          | 2009-01-13T01:08:05.680 | 2009 | 1     |
| 129 | false    | 9           | 9            | 9       | 90         | 9.9       | 90.89999999999999  | 01/13/09        | 9          | 2009-01-13T01:09:05.760 | 2009 | 1     |
| 130 | true     | 0           | 0            | 0       | 0          | 0.0       | 0.0                | 01/14/09        | 0          | 2009-01-14T01:10:05.850 | 2009 | 1     |
| 131 | false    | 1           | 1            | 1       | 10         | 1.1       | 10.1               | 01/14/09        | 1          | 2009-01-14T01:11:05.850 | 2009 | 1     |
| 132 | true     | 2           | 2            | 2       | 20         | 2.2       | 20.2               | 01/14/09        | 2          | 2009-01-14T01:12:05.860 | 2009 | 1     |
| 133 | false    | 3           | 3            | 3       | 30         | 3.3       | 30.299999999999997 | 01/14/09        | 3          | 2009-01-14T01:13:05.880 | 2009 | 1     |
| 134 | true     | 4           | 4            | 4       | 40         | 4.4       | 40.4               | 01/14/09        | 4          | 2009-01-14T01:14:05.910 | 2009 | 1     |
| 135 | false    | 5           | 5            | 5       | 50         | 5.5       | 50.5               | 01/14/09        | 5          | 2009-01-14T01:15:05.950 | 2009 | 1     |
| 136 | true     | 6           | 6            | 6       | 60         | 6.6       | 60.599999999999994 | 01/14/09        | 6          | 2009-01-14T01:16:06     | 2009 | 1     |
| 137 | false    | 7           | 7            | 7       | 70         | 7.7       | 70.7               | 01/14/09        | 7          | 2009-01-14T01:17:06.600 | 2009 | 1     |
| 138 | true     | 8           | 8            | 8       | 80         | 8.8       | 80.8               | 01/14/09        | 8          | 2009-01-14T01:18:06.130 | 2009 | 1     |
| 139 | false    | 9           | 9            | 9       | 90         | 9.9       | 90.89999999999999  | 01/14/09        | 9          | 2009-01-14T01:19:06.210 | 2009 | 1     |
| 140 | true     | 0           | 0            | 0       | 0          | 0.0       | 0.0                | 01/15/09        | 0          | 2009-01-15T01:20:06.300 | 2009 | 1     |
| 141 | false    | 1           | 1            | 1       | 10         | 1.1       | 10.1               | 01/15/09        | 1          | 2009-01-15T01:21:06.300 | 2009 | 1     |
| 142 | true     | 2           | 2            | 2       | 20         | 2.2       | 20.2               | 01/15/09        | 2          | 2009-01-15T01:22:06.310 | 2009 | 1     |
| 143 | false    | 3           | 3            | 3       | 30         | 3.3       | 30.299999999999997 | 01/15/09        | 3          | 2009-01-15T01:23:06.330 | 2009 | 1     |
| 144 | true     | 4           | 4            | 4       | 40         | 4.4       | 40.4               | 01/15/09        | 4          | 2009-01-15T01:24:06.360 | 2009 | 1     |
| 145 | false    | 5           | 5            | 5       | 50         | 5.5       | 50.5               | 01/15/09        | 5          | 2009-01-15T01:25:06.400 | 2009 | 1     |
| 98  | true     | 8           | 8            | 8       | 80         | 8.8       | 80.8               | 01/10/09        | 8          | 2009-01-10T00:38:04.330 | 2009 | 1     |
| 99  | false    | 9           | 9            | 9       | 90         | 9.9       | 90.89999999999999  | 01/10/09        | 9          | 2009-01-10T00:39:04.410 | 2009 | 1     |
| 100 | true     | 0           | 0            | 0       | 0          | 0.0       | 0.0                | 01/11/09        | 0          | 2009-01-11T00:40:04.500 | 2009 | 1     |
| 101 | false    | 1           | 1            | 1       | 10         | 1.1       | 10.1               | 01/11/09        | 1          | 2009-01-11T00:41:04.500 | 2009 | 1     |
| 102 | true     | 2           | 2            | 2       | 20         | 2.2       | 20.2               | 01/11/09        | 2          | 2009-01-11T00:42:04.510 | 2009 | 1     |
| 103 | false    | 3           | 3            | 3       | 30         | 3.3       | 30.299999999999997 | 01/11/09        | 3          | 2009-01-11T00:43:04.530 | 2009 | 1     |
| 104 | true     | 4           | 4            | 4       | 40         | 4.4       | 40.4               | 01/11/09        | 4          | 2009-01-11T00:44:04.560 | 2009 | 1     |
| 105 | false    | 5           | 5            | 5       | 50         | 5.5       | 50.5               | 01/11/09        | 5          | 2009-01-11T00:45:04.600 | 2009 | 1     |
| 106 | true     | 6           | 6            | 6       | 60         | 6.6       | 60.599999999999994 | 01/11/09        | 6          | 2009-01-11T00:46:04.650 | 2009 | 1     |
| 107 | false    | 7           | 7            | 7       | 70         | 7.7       | 70.7               | 01/11/09        | 7          | 2009-01-11T00:47:04.710 | 2009 | 1     |
| 108 | true     | 8           | 8            | 8       | 80         | 8.8       | 80.8               | 01/11/09        | 8          | 2009-01-11T00:48:04.780 | 2009 | 1     |
| 109 | false    | 9           | 9            | 9       | 90         | 9.9       | 90.89999999999999  | 01/11/09        | 9          | 2009-01-11T00:49:04.860 | 2009 | 1     |
| 110 | true     | 0           | 0            | 0       | 0          | 0.0       | 0.0                | 01/12/09        | 0          | 2009-01-12T00:50:04.950 | 2009 | 1     |
| 111 | false    | 1           | 1            | 1       | 10         | 1.1       | 10.1               | 01/12/09        | 1          | 2009-01-12T00:51:04.950 | 2009 | 1     |
| 112 | true     | 2           | 2            | 2       | 20         | 2.2       | 20.2               | 01/12/09        | 2          | 2009-01-12T00:52:04.960 | 2009 | 1     |
| 113 | false    | 3           | 3            | 3       | 30         | 3.3       | 30.299999999999997 | 01/12/09        | 3          | 2009-01-12T00:53:04.980 | 2009 | 1     |
| .                                                                                                                                                                           |
| .                                                                                                                                                                           |
| .                                                                                                                                                                           |
+-----+----------+-------------+--------------+---------+------------+-----------+--------------------+-----------------+------------+-------------------------+------+-------+
7300 row(s) fetched. (First 40 displayed. Use --maxrows to adjust)
Elapsed 0.015 seconds.
```

**Review by alamb — APPROVED** [review]
Looks really nice to me now -- thank you @Weijun-H
